File: src/java/org/objectweb/asm/commons/cfg/CallVisitor.java
package org.objectweb.asm.commons.cfg;

import org.objectweb.asm.Handle;
import org.objectweb.asm.MethodVisitor;
import org.objectweb.asm.commons.cfg.graph.CallGraph;
import org.objectweb.asm.tree.MethodInsnNode;
import org.objectweb.asm.tree.MethodNode;

/**
 * @author Tyler Sedlar
 */
public class CallVisitor extends MethodVisitor {

    public final CallGraph graph = new CallGraph();

    private MethodNode mn;

    public void visit(MethodNode mn) {
        this.mn = mn;
        mn.accept(this);
    }

    @Override
    public void visitMethodInsn(MethodInsnNode min) {
        graph.addMethodCall(mn.handle, new Handle(0, min.owner, min.name, min.desc));
    }
}

File: src/kotlin/kt/osrs/analysis/tree/flow/CallVisitor.kt
package kt.osrs.analysis.tree.flow

import kt.osrs.analysis.tree.flow.graph.CallGraph
import org.objectweb.asm.Handle
import org.objectweb.asm.MethodVisitor
import org.objectweb.asm.tree.MethodInsnNode
import org.objectweb.asm.tree.MethodNode

/**
 * @author Tyler Sedlar
 */
class CallVisitor : MethodVisitor() {

    val graph = CallGraph()

    private var mn: MethodNode? = null

    fun visit(mn: MethodNode) {
        this.mn = mn
        mn.accept(this)
    }

    override fun visitMethodInsn(min: MethodInsnNode) {
        graph.addMethodCall(mn!!.handle, Handle(0, min.owner, min.name, min.desc))
    }
}